tests/daemon/test-aggregate-tally.c

```diff
@@ -0,0 +1,254 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*- */
+
+/* Copyright 2021 Endless OS Foundation, LLC */
+
+/*
+ * This file is part of eos-event-recorder-daemon.
+ *
+ * eos-event-recorder-daemon is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * eos-event-recorder-daemon is distributed in the hope that it will be
+ * useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with eos-event-recorder-daemon.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+
+#include "emer-aggregate-tally.h"
+
+#include "shared/metrics-util.h"
+
+#include <glib.h>
+#include <glib/gstdio.h>
+#include <uuid/uuid.h>
+
+#include <eosmetrics/eosmetrics.h>
+
+#define UUID_LENGTH (sizeof (uuid_t) / sizeof (guchar))
+
+// Generated with uuidgen
+
+static const char *uuids[] = {
+  "41d45e08-5e72-4c43-8cbf-ef37bb4411a4",
+  "03177773-7513-4866-ae97-bb935f2c5384",
+  "a692ce9c-8684-4d6b-97d5-07f39e0a8560",
+};
+
+struct AggregateEvent
+{
+  guint32 unix_user_id;
+  GVariant *event_id;
+  GVariant *aggregate_key;
+  GVariant *payload;
+};
+
+struct Fixture
+{
+  EmerAggregateTally *tally;
+  GFile *tally_folder;
+};
+
+static GVariant *
+event_id_to_variant (const char *event_id)
+{
+  GVariantBuilder event_id_builder;
+  uuid_t parsed_event_id;
+
+  if (uuid_parse (event_id, parsed_event_id) != 0)
+    return NULL;
+
+  get_uuid_builder (parsed_event_id, &event_id_builder);
+
+  return g_variant_new ("ay", &event_id_builder);
+}
+
+static struct AggregateEvent *
+create_aggregate_event (guint32     unix_user_id,
+                        const char *event_id,
+                        GVariant   *aggregate_key,
+                        GVariant   *payload)
+{
+  struct AggregateEvent *event;
+
+  event = g_new0 (struct AggregateEvent, 1);
+  event->unix_user_id = unix_user_id;
+  event->event_id = g_variant_ref_sink (event_id_to_variant (event_id));
+  event->aggregate_key = g_variant_ref_sink (aggregate_key);
+  event->payload = g_variant_ref_sink (payload);
+
+  return event;
+}
+
+static void
+aggregate_event_free (struct AggregateEvent *event)
+{
+  g_clear_pointer (&event->event_id, g_variant_unref);
+  g_clear_pointer (&event->aggregate_key, g_variant_unref);
+  g_clear_pointer (&event->payload, g_variant_unref);
+  g_free (event);
+}
+
+static void
+teardown (struct Fixture *fixture,
+          gconstpointer   dontuseme)
+{
+  g_clear_object (&fixture->tally);
+
+  if (fixture->tally_folder)
+    {
+      delete_folder (fixture->tally_folder, NULL, NULL);
+      g_clear_object (&fixture->tally_folder);
+    }
```

Comment:
With G_TEST_OPTION_ISOLATE_DIRS this teardown code is not needed. GLib will delete the temporary directory for you. As a result `delete_folder` is also not needed.